### PR TITLE
add and implement custom exceptions to service

### PR DESCRIPTION
**Pull request by STOzaki**
This will have the services throw custom errors for the menu to handle and alert the user, minimizing silent failures.

### CODE REVIEW

File: src/com/lms/customExceptions/TransactionException.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.lms.customExceptions;
+
+@SuppressWarnings("serial")
+public class TransactionException extends Exception {
+
+	public TransactionException(String errorMessage) {
```
Comment (kingjon3377) [inline src/com/lms/customExceptions/TransactionException.java]: Will we ever want to create an instance that isn't one of the subclasses? If not, make this constructor `protected` instead of `public`.

File: src/com/lms/service/AdministratorService.java [hunk 2]
```diff
@@ -27,7 +30,7 @@
 	 * @param publisher the publisher of the book
 	 * @return the newly-created book
 	 */
-	Book createBook(String title, Author author, Publisher publisher) throws SQLException;
+	Book createBook(String title, Author author, Publisher publisher) throws InsertException;
```
Comment (kingjon3377) [inline src/com/lms/service/AdministratorService.java]: Throughout, I would declare each method (in the interface) `throws TransactionException` rather than the specific subclass, just in case at some point an implementation needs to use one of the others.

File: src/com/lms/service/BorrowerService.java [hunk 3]
```diff
@@ -46,15 +48,13 @@
 	 * the given borrower from the given branch, and the book is not overdue, remove
 	 * the loan from the database and return true. If it is overdue, return false.
 	 *
-	 * <p>TODO: What to do when no matching loan exists?
-	 *
 	 * @param borrower the borrower returning the book
 	 * @param book the book being returned
 	 * @param branch the branch from which it was borrowed
-	 * @param dueDate TODO: document this once author clarifies its purpose
```
Comment (kingjon3377) [inline src/com/lms/service/BorrowerService.java:54]: The changes to the parameter name and return type of this method should be in separate commits from the exception-type changes.

**Review by kingjon3377 — COMMENTED** [review]
I recommend rebasing to split the `returnBook` semantic changes into a separate commit or commits, and declaring `throws TransactionException` rather than the specific subclass, but other than that 👍 .

File: src/com/lms/customExceptions/TransactionException.java (hunk 1 above)
Comment (STOzaki) [reply src/com/lms/customExceptions/TransactionException.java]: I will actually do as your other comment suggests and just call the super class instead.

**Review by al-amine — APPROVED** [review]
Approved

File: src/com/lms/customExceptions/TransactionException.java (hunk 1 above)
Comment (kingjon3377) [reply src/com/lms/customExceptions/TransactionException.java]: The two changes are unrelated to each other. But do as you will.

**Review by kingjon3377 — APPROVED** [review]
Looks good to me!

File: src/com/lms/customExceptions/TransactionException.java (hunk 1 above)
Comment (STOzaki) [reply src/com/lms/customExceptions/TransactionException.java]: I will change that. You are correct.
